discovery-provider/plugins/notifications/src/processNotifications/mappers/reaction.ts

```diff
@@ -0,0 +1,95 @@
+import { Knex } from 'knex'
+import { NotificationRow, UserRow } from '../../types/dn'
+import { ReactionNotification } from '../../types/notifications'
+import { BaseNotification, Device } from './base'
+import { sendPushNotification } from '../../sns'
+import { ResourceIds, Resources } from '../../email/notifications/renderEmail'
+import { capitalize } from '../../email/notifications/components/utils'
+import { formatWei } from '../../utils/format'
+
+type ReactionNotificationRow = Omit<NotificationRow, 'data'> & { data: ReactionNotification }
+export class Reaction extends BaseNotification<ReactionNotificationRow> {
+
+  reactedTo: string
+  reactionType: string
+  reactionValue: number
+  senderWallet: string
+  receiverUserId: number
+  senderUserId: number
+  tipAmount: string
+
+  constructor(dnDB: Knex, identityDB: Knex, notification: ReactionNotificationRow) {
+    super(dnDB, identityDB, notification)
+    const userIds: number[] = this.notification.user_ids!
+    this.reactedTo = this.notification.data.reacted_to
+    this.reactionType = this.notification.data.reaction_type
+    this.reactionValue = this.notification.data.reaction_value
+    this.senderWallet = this.notification.data.sender_wallet
+    this.receiverUserId = this.notification.data.receiver_user_id
+    this.senderUserId = this.notification.data.sender_user_id
+    this.tipAmount = this.notification.data.tip_amount
+  }
+
+  async pushNotification() {
+    const res: Array<{ user_id: number, name: string, is_deactivated: boolean }> = await this.dnDB.select('user_id', 'name', 'is_deactivated')
+      .from<UserRow>('users')
+      .where('is_current', true)
+      .whereIn('user_id', [this.receiverUserId, this.senderUserId])
+    const users = res.reduce((acc, user) => {
+      acc[user.user_id] = { name: user.name, isDeactivated: user.is_deactivated }
+      return acc
+    }, {} as Record<number, { name: string, isDeactivated: boolean }>)
+
+    if (users?.[this.senderUserId]?.isDeactivated) {
+      return
+    }
```

Comment:
another nit - looks like we have to do this across all notifications so wondering if there's a place they could all share this code? maybe extracting it to a function that could be in some utils? or as part of the base class?

Reply:
Yes, it is a todo to clean up and DRY but will leave in for now